### PR TITLE
Add support for sky

### PR DESCRIPTION
**Pull request by iche033**
depends on:
* ~~ignitionrobotics/ign-rendering#168~~
* ~~osrf/sdformat#397~~

Users can now enable a sky in the scene through the `<sky>` SDF element. It's currently only a static skybox background and sky properties like wind speed, humidity, etc are not supported yet. Only works in ogre2 rendering engine. 

To test:

```
ign gazebo -v 4 sky.sdf
```

sky_ign-gazebo

Known issue:

Sky is not visible to an RGBD camera yet.

### CODE REVIEW

**Review by chapulina — APPROVED** [review]
Looks sunny! :sun_with_face: 

Is there a plan to support Ogre 1 and the sky properties in the near future?

File: examples/worlds/sky.sdf [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+      <background>0.8 0.8 0.8</background>
+      <sky></sky>
+    </scene>
+
+    <light type="directional" name="sun">
+      <cast_shadows>true</cast_shadows>
+      <pose>0 0 10 0 0 0</pose>
+      <diffuse>0.8 0.8 0.8 1</diffuse>
+      <specular>0.8 0.8 0.8 1</specular>
+      <direction>-0.5 0.1 -0.9</direction>
```
[the remark this answers is not in the document]
Comment (iche033) [reply examples/worlds/sky.sdf]: updated light direction fccdf5a :sun_behind_large_cloud:

File: src/gui/plugins/scene3d/Scene3D.cc [hunk 2]
```diff
@@ -2147,6 +2147,11 @@ void Scene3D::LoadConfig(const tinyxml2::XMLElement *_pluginElem)
       this->dataPtr->renderUtil->SetBackgroundColor(bgColor);
     }
 
+    if (_pluginElem->FirstChildElement("sky"))
+    {
```
[the remark this answers is not in the document]
Comment (iche033) [reply src/gui/plugins/scene3d/Scene3D.cc:2151]: added warning. fccdf5a

**Comment by iche033** [comment]
> Is there a plan to support Ogre 1 and the sky properties in the near future?

I think for ogre1, we can just port over the skyx code from gazebo classic. As for supporting other sky properties in ogre2, maybe we can see if it's possible to port the skyx implementation to ogre2.

**Review by nkoenig — APPROVED** [review]

**Comment by iche033** [comment]
sdformat 10.1.0 was released with sky DOm so I think we can merge this now.

**Comment by chapulina** [comment]
> Has anyone looked into this more

I believe no one has looked into this yet, or has plans to look into it in the near future

**Pull request merged**